cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -0,0 +1,140 @@
+	tagsuite.Suite
+}
+
+func (suite *SvcIntegrationTestSuite) TestStartStop() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("start"))
+	cp.Expect("Starting")
+	cp.ExpectExitCode(0)
+	time.Sleep(500 * time.Millisecond) // wait for service to start up
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Checking")
+
+	// Verify it created a socket file.
+	sockFile := svcctl.NewIPCSockPathFromGlobals().String()
+	suite.True(fileutils.TargetExists(sockFile))
+
+	// Verify the server is running on its reported port.
+	cp.ExpectRe("Port:\\s+:\\d+")
+	portRe := regexp.MustCompile("Port:\\s+:(\\d+)")
+	port := portRe.FindStringSubmatch(cp.TrimmedSnapshot())[1]
+	_, err := net.Listen("tcp", "localhost:"+port)
+	suite.Error(err)
+
+	// Verify it created and wrote to its reported log file.
+	cp.ExpectRe("Log:\\s+.+?\\.log")
+	logRe := regexp.MustCompile("Log:\\s+(.+?\\.log)")
+	logFile := logRe.FindStringSubmatch(cp.TrimmedSnapshot())[1]
+	suite.True(fileutils.FileExists(logFile))
+	suite.True(len(fileutils.ReadFileUnsafe(logFile)) > 0)
+
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("stop"))
+	cp.Expect("Stopping")
+	cp.ExpectExitCode(0)
+	time.Sleep(500 * time.Millisecond) // wait for service to stop
+
+	// Verify it deleted its socket file.
+	suite.False(fileutils.TargetExists(sockFile))
+}
+
+func (suite *SvcIntegrationTestSuite) TestSignals() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	// SIGINT (^C)
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("foreground"))
+	cp.Expect("Starting")
+	cp.Signal(syscall.SIGINT)
+	cp.Expect("caught a signal: interrupt")
+	cp.ExpectNotExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	sockFile := svcctl.NewIPCSockPathFromGlobals().String()
+	suite.False(fileutils.TargetExists(sockFile))
+
+	// SIGTERM
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("foreground"))
+	cp.Expect("Starting")
+	cp.Signal(syscall.SIGTERM)
+	suite.NotContains(cp.TrimmedSnapshot(), "caught a signal")
+	cp.ExpectNotExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	suite.False(fileutils.TargetExists(sockFile))
+}
+
+func (suite *SvcIntegrationTestSuite) TestSingleSvc() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	oldCount := suite.GetNumStateSvcProcesses() // may be non-zero due to non-test state-svc processes
+	for i := 1; i <= 10; i++ {
+		go func() {
+			ts.SpawnCmdWithOpts(ts.Exe, e2e.WithArgs("--version"))
+		}()
+		time.Sleep(50 * time.Millisecond)
```

Comment:
```suggestion
```

This shouldn't be necessary. If not having this breaks the test then they are correctly breaking and we need to address the underlying issue.